kedro/templates/project/hooks/post_gen_project.py

```diff
@@ -0,0 +1,106 @@
+import json
+from pathlib import Path
+import shutil
+
+from kedro.templates.project.hooks.utils import parse_add_ons_input
+
+# Get the selected add-ons from cookiecutter
+selected_add_ons = "{{ cookiecutter.add_ons }}"
+
+# Parse the add-ons to get a list
+selected_add_ons_list = parse_add_ons_input(selected_add_ons)
+
+current_dir = Path.cwd()
+requirements_file_path = current_dir / "src/requirements.txt"
+pyproject_file_path = current_dir / "pyproject.toml"
+pyproject_src_file_path = current_dir / "src/pyproject.toml"
+
+if "1" not in selected_add_ons_list:  # If Linting not selected
+    setup_cfg_path = current_dir / "setup.cfg"
```

Comment:
This unlinks a file that isn't part of the current project template 🤔 

Additionally, we don't recommend using `setup.cfg` in our [docs](https://docs.kedro.org/en/latest/development/linting.html), instead using `pyproject.toml`.